src/app/clusters/bindings/BindingManager.h

```diff
@@ -70,6 +72,16 @@ class BindingManager
 
     void RegisterBoundDeviceChangedHandler(BoundDeviceChangedHandler handler) { mBoundDeviceChangedHandler = handler; }
 
+    /*
+     * Registers handler that will be called when context used in NotifyBoundClusterChanged will not be needed and could be
+     * released.
```

Comment:
I think this should clearly document that it might be called before `NotifyBoundClusterChanged` returns....

Reply:
After reading that comment I realized that connection handler can be called even before `NotifyBoundClusterChanged` end and it will result in deleting the context used in further iterations. I decided to wrap the context into the type that will count consumers and take care of releasing the context only if no consumer left. Please see the last commit.